agent/acl_test.go

```diff
@@ -130,6 +130,19 @@ func (a *TestACLAgent) ResolveTokenAndDefaultMeta(secretID string, entMeta *stru
 	return authz, err
 }
 
+func (a *TestACLAgent) ResolveIdentityFromToken(secretID string) (bool, structs.ACLIdentity, error) {
```

Comment:
This is almost certainly not useful test behavior, but I have no idea how I should be testing this x)

I've was getting an infinite loop where I accidentally dispatch right back to the delegate, turns out I was implementing the agent interface... against itself? Still don't actually understand it. But the actual solution was implementing ResolveIdentityFromToken on the TestACLAgent.

Reply:
This was the right thing to do. `TestACLAgent` is quite special. The whole reason it exists is to be able to test the code in agent/acl.go without having to start a full `TestAgent` which would run various go routines etc. It also allows us to intercept any of the delegate functions of the agent and to inject various conditions that a normally functioning agent delegate (either the `Client` or `Server` type) would not be able to do but that are theoretically possible according to the interface being used.

It is a bit circular in that the `TestACLAgent` holds onto an `Agent` and sets itself up as the agents delegate. We could have instead created a second struct to implement the delegate and have the `TestACLAgent` hold on to both the `Agent` and that other struct.

Reply:
OK, yeah I think that my mistake was simply not understanding how it worked -- I don't see any particular issue with the `TestACLAgent`, just my usage of it.